File: sanidad-backend/src/main/java/com/farmacia/sanidadbackend/dto/LoteRequest.java
package com.farmacia.sanidadbackend.dto;

import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;
import java.util.List;

@Getter
@Setter
public class LoteRequest {

    private LocalDate fechaFabricacion;

    private LocalDate fechaVencimiento;

    @NotNull(message = "El proveedor es obligatorio")
    private Long proveedorId;

    private String factura;

    @NotEmpty(message = "Debe incluir al menos un medicamento")
    private List<LoteDetalleRequest> detalles;
}

File: SanidadMovil/app/src/main/java/com/sanidad/movil/data/remote/dto/LoteRequest.kt
package com.sanidad.movil.data.remote.dto

data class LoteRequest(
    val fechaFabricacion: String? = null, // ISO date
    val fechaVencimiento: String? = null,
    val proveedorId: Long,
    val factura: String? = null,
    val detalles: List<LoteDetalleRequest>
)